jetstream/core/orchestrator.py

```diff
@@ -795,6 +792,44 @@ def _detokenize_thread(self, idx: int):
         slot, active_request = data
         my_live_requests[slot] = active_request
 
+  def _prefill_detokenize_thread(self, idx: int):
```

Comment:
I though we already had prefill detokenize thread. Do current jetstream (before this pr) always return prefill token (fist token) after first decode step?

Reply:
We only had detokenize thread that combined prefill detokenize and decode detokenize. The problem is that we have `jax.block_until_ready()` blocking the thread waiting for the prefill token or decode token copy to host async, so putting them in 1 thread would make the TTFT slow. JetStream returns prefill token in prefill thread (after prefill step generating the first token).

Reply:
sounds good, thanks for sharing insights!